targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF53/i2c_api.c

```diff
@@ -121,20 +318,75 @@ void i2c_frequency(i2c_t *obj, int hz)
             MBED_WARNING1(MBED_MAKE_ERROR(MBED_MODULE_DRIVER_SERIAL, MBED_ERROR_CODE_UNSUPPORTED), "frequency not supported", hz);
             break;
     }
+}
+
+/**
+ * @brief      Reconfigure driver.
+ *
+ *             If the peripheral is enabled, it will be disabled first. All
+ *             registers are cleared to their default values unless replaced
+ *             by new configuration.
+ *
+ * @param      obj           The object
+ */
+static void i2c_configure_driver_instance(i2c_t *obj_)
+{
+    struct i2c_s *obj = GET_I2C_S(obj_);
+
+    /* If the peripheral is running, disable and uninitialize it.*/
+    if (is_master_active(obj_))
+        deinit_master(obj_);
 
-    nrfx_twim_init(&obj->instance, &obj->config, i2c_event_handler, obj);
-    nrfx_twim_enable(&obj->instance);
+#if DEVICE_I2CSLAVE
+    if (is_slave_active(obj_))
+        deinit_slave(obj_);
+
+    /* Configure slave driver if device is slave */
+    if (obj->is_slave) {
+        nrfx_twis_config_t twis_config =
+            NRFX_TWIS_DEFAULT_CONFIG(obj->config.scl, obj->config.sda,
+                                     obj->slave_addr >> 1);
+
+        /* FIXME: add error handling */
+        nrfx_twis_init(&obj->slave_instance, &twis_config, twis_event_handler);
+        nrfx_twis_enable(&obj->slave_instance);
+        nrfx_twis_in_use[obj->slave_instance.drv_inst_idx] = true;
+
+        /* FIXME: use array of i2c_s ptr instead? */
+        global_i2c_s = obj;
+        return;
+    }
+#endif // DEVICE_I2CSLAVE
+
+    /* Configure driver as master if device is master */
+    if (obj->instance.p_twim) {
+        nrfx_twim_init(&obj->instance, &obj->config, i2c_event_handler, obj_);
+        nrfx_twim_enable(&obj->instance);
+        nrfx_twim_in_use[obj->instance.drv_inst_idx] = true;
+    }
 }
 
-int i2c_start(i2c_t *obj)
+int i2c_start(i2c_t *obj_)
 {
+    struct i2c_s *obj = GET_I2C_S(obj_);
+
     obj->length = 0;
     obj->address = 0;
+
     return 0;
 }
 
-int i2c_byte_write(i2c_t *obj, int data)
+int i2c_byte_write(i2c_t *obj_, int data)
 {
+    struct i2c_s *obj = GET_I2C_S(obj_);
+
+    /* return if master instance is invalid */
+    if (!obj->instance.p_twim)
+        return 0;
```

Comment:
Oh yeah, btw, for these errors, there were two predefined ones (BUS_BUSY or NO_SLAVE)  so I just chose the NO_SLAVE for now. It is also used when trying to read from a master bus as far as I could see. It would probably be more correct to extend the errors to also have an appropriate error for these cases